test/unit-tests/source_fetch/source_fetch.ml

```diff
@@ -0,0 +1,38 @@
+open Stdune
+module Fetch = Dune_pkg.Fetch
+module Scheduler = Dune_engine.Scheduler
+
+let main () =
+  let open Fiber.O in
+  let url = OpamUrl.of_string "https://xivilization.net/~marek/kimiko.txt" in
```

Comment:
Can we make this test not depend the network? Also, I'd like to see some tests with git repos, tarballs, compressed tarballs, etc.

Reply:
Yeah, a local http server is a decent option.

Reply:
Added the most basic local HTTP server that speaks just enough HTTP to serve one request and exit.